ot3_state_manager/ot3_state_manager/messages.py

```diff
@@ -137,24 +202,29 @@ def _parse_message(message_bytes: bytes) -> Message:
 
 def handle_message(data: bytes, ot3_state: OT3State) -> bytes:
     """Function to handle incoming message, react to it accordingly, and respond."""
-    error_response: Optional[str] = None
+    response = Response(content=None, is_error=False)
     try:
         message = _parse_message(data)
     except ValueError as err:
-        error_response = f"{err.args[0]}"
+        response.is_error = True
+        response.content = f"{err.args[0]}"
     except:  # noqa: E722
-        error_response = "Unhandled Exception"
+        response.is_error = True
+        response.content = "Unhandled Exception"
     else:
         if isinstance(message, MoveMessage):
             ot3_state.pulse(message.axis, message.direction)
+            response.content = data.decode()
```

Comment:
maybe a different pr, but i'd recommend pushing this behavior down into the classes somehow - give them a `handle` function so we don't need a big if/elif block that we can't really statically verify

Reply:
7586e06b0aa8bfedfb2b8d9946e7750010978aaa